cmd/thanos/main.go

```diff
@@ -355,24 +356,27 @@ func metricHTTPListenGroup(g *run.Group, logger log.Logger, reg *prometheus.Regi
 
 // defaultHTTPListener starts a run.Group that servers HTTP endpoint with default endpoints providing Prometheus metrics,
 // profiling and liveness/readiness probes.
-func defaultHTTPListener(g *run.Group, logger log.Logger, reg *prometheus.Registry, httpBindAddr string, readinessProber *prober.Prober) error {
+func defaultHTTPListener(comp component.Component, g *run.Group, logger log.Logger, reg *prometheus.Registry, router http.Handler, httpBindAddr string, readinessProber *prober.Prober) error {
 	mux := http.NewServeMux()
 	registerMetrics(mux, reg)
 	registerProfile(mux)
 	readinessProber.RegisterInMux(mux)
+	if router != nil {
+		mux.Handle("/", router)
+	}
 
 	l, err := net.Listen("tcp", httpBindAddr)
 	if err != nil {
 		return errors.Wrap(err, "listen metrics address")
 	}
 
 	g.Add(func() error {
-		level.Info(logger).Log("msg", "listening for metrics", "address", httpBindAddr)
+		level.Info(logger).Log("msg", fmt.Sprintf("listening for %s and metrics", comp.String()), "address", httpBindAddr)
```

Comment:
I would not change this way. We aim for structured logging. While it might be bit confusing for human we want to spread variables as separate args like `address` `component` `msg`... what do you think? Then we don't need `fmt.Sprintf(`